nixos/modules/services/cluster/kubernetes/default.nix

```diff
@@ -6,6 +6,13 @@ let
   cfg = config.services.kubernetes;
   opt = options.services.kubernetes;
 
+  renderArg = v:
+    if isList v
+    then concatStringsSep "," v
+    else if isBool v
+    then boolToString v
+    else toString v;
```

Comment:
Sure we want a generic fallthrough to `toString` here over an error? 

Also I tend to do these like "we have case statements at home," don't know if you care :smile_cat: 

```
  renderArg = v: let
    cases = {
      string = toString;
      float = toString;
      int = toString;
      list = lib.concatMapStringsSep "," renderArg;
    };
    type = builtins.typeOf v;
    default = throw "Unhandled type ${type} in renderArg";
  in (cases.${type} or default) v;
```

What I like about naming this is that each of the freeForm option types should be covered by the switches, and we should be good (or have at least thought about each option)